diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt

```diff
@@ -162,7 +158,13 @@ class HeaderCommentRule(private val configRules: List<RulesConfig>) : Rule("head
                 copyrightWords.any { commentNode.text.contains(it, ignoreCase = true) }
             }
         if (isWrongCopyright || isMissingCopyright || isCopyrightInsideKdoc) {
-            HEADER_MISSING_OR_WRONG_COPYRIGHT.warnAndFix(configRules, emitWarn, isFixMode, fileName, node.startOffset, node) {
+            val freeText = when {
```

Comment:
What if several flags are true? Maybe better to do string concatenation?

Reply:
I think it doesn't affect meaning:
* `isMissingCopyright` means block comment doesn't exist
* `isWrongCopyright` means it exists but doesn't have keywords
* `isCopyrightInsideKdoc` means block comment doesn't (which makes `isMissingCopyright` `true`), but there is copyright information inside KDoc, which makes description of `isMissingCopyright` redundant here

Reply:
But you are right, it still needs some refinement